edgelet/iotedge/src/check/mod.rs

```diff
@@ -727,6 +728,65 @@ fn container_engine(check: &mut Check) -> Result<CheckResult, failure::Error> {
     Ok(CheckResult::Ok)
 }
 
+fn container_engine_network(check: &mut Check) -> Result<CheckResult, failure::Error> {
+    const MESSAGE: &str =
+        "Container engine is not configured for IPv6 communication.\n\
+         Please see https://docs.docker.com/v17.09/engine/userguide/networking/default_network/ipv6 \
+         for a guide on how to enable IPv6 support.\n";
+
+    #[derive(serde_derive::Deserialize)]
+    struct DaemonConfig {
+        ipv6: Option<bool>,
+    }
+
+    let is_edge_ipv6_configured = &check.settings.as_ref().map_or(false, |settings| {
+        let moby_network = settings.moby_runtime().network();
+        if let MobyNetwork::Network(network) = moby_network {
+            network.ipv6().unwrap_or_default()
+        } else {
+            false
+        }
+    });
+
+    let daemon_config_file = File::open(&check.container_engine_config_path)
+        .with_context(|_| {
+            format!(
+                "Could not open container engine config file {}",
+                check.container_engine_config_path.display(),
+            )
+        })
+        .context(MESSAGE);
+    let daemon_config_file = match daemon_config_file {
+        Ok(daemon_config_file) => daemon_config_file,
+        Err(err) => {
+            if *is_edge_ipv6_configured {
+                return Ok(CheckResult::Fatal(err.into()));
+            } else {
+                return Ok(CheckResult::Ok);
+            }
+        }
+    };
+    let daemon_config: DaemonConfig = serde_json::from_reader(daemon_config_file)
+        .with_context(|_| {
+            format!(
+                "Could not parse container engine config file {}",
+                check.container_engine_config_path.display(),
+            )
+        })
+        .context(MESSAGE)?;
+
+    if daemon_config.ipv6.unwrap_or_default() {
```

Comment:
```rust
match (daemon_config.ipv6.unwrap_or_default(), is_edge_ipv6_configured) {
    (true, _) if cfg!(windows) => Err(<not supported on Windows>),
    (false, true) => Err(<MESSAGE>),
    _ => Ok(CheckResult::Ok),
}
```

Reply:
What's the general idea behind returning Fatal errors or Err?
Also, the way it is right now, we're returning CheckResult::Skipped if the config.yaml doesn't have IPv6 enabled and it's the Unix environment. I kept it that way cause I didn't want the check to show up in that case cause the customer never even tried to enable IPv6 with Edge.

Reply:
Reading the comments that describe each of the CheckResult variants, it appears that we return Fatal in case no other checks should be performed if one of the checks returns this error. In that case, shouldn't we return a Fatal error for both the cases -

1. daemon.json has IPv6 set to `true` on Windows.
2. IPv6 is enabled in config.yaml and not in daemon.json.

Either of these conditions would render Edge completely broken as no modules can be created.

Reply:
No.

Reply:
`Fatal` is for when it doesn't make sense to run other checks. That's why it's only used for two cases: where the user needs to rerun the command with privs, or where the OS is fundamentally unsupported.

Reply:
Also, `Skipped` means the check was skipped because previous checks failed. This is not the case the way you were using it.

If you want to hide the check, the variant for that is `Ignored`. Yes you can add a `(false, false)` arm that returns `Ignored` if you want.

Reply:
Cool, I'll use Ignored then and return Err.